steering_controllers/src/steering_controllers.cpp

```diff
@@ -0,0 +1,489 @@
+  {
+    command_interfaces_config.names.push_back(
+      params_.front_steer_names[i] + "/" + hardware_interface::HW_IF_POSITION);
+  }
+  return command_interfaces_config;
+}
+
+controller_interface::InterfaceConfiguration SteeringControllers::state_interface_configuration()
+  const
+{
+  controller_interface::InterfaceConfiguration state_interfaces_config;
+  state_interfaces_config.type = controller_interface::interface_configuration_type::INDIVIDUAL;
+
+  state_interfaces_config.names.reserve(nr_state_itfs_);
+  const auto rear_wheel_feedback = params_.position_feedback ? hardware_interface::HW_IF_POSITION
+                                                             : hardware_interface::HW_IF_VELOCITY;
+
+  for (size_t i = 0; i < params_.front_steer_names.size(); i++)
+  {
+    state_interfaces_config.names.push_back(
+      params_.rear_wheel_names[i] + "/" + rear_wheel_feedback);
+  }
+
+  for (size_t i = 0; i < params_.front_steer_names.size(); i++)
+  {
+    state_interfaces_config.names.push_back(
+      params_.front_steer_names[i] + "/" + hardware_interface::HW_IF_POSITION);
+  }
+
+  return state_interfaces_config;
+}
+
+std::vector<hardware_interface::CommandInterface>
+SteeringControllers::on_export_reference_interfaces()
+{
+  reference_interfaces_.resize(nr_ref_itfs_, std::numeric_limits<double>::quiet_NaN());
+
+  std::vector<hardware_interface::CommandInterface> reference_interfaces;
+  reference_interfaces.reserve(nr_ref_itfs_);
+
+  reference_interfaces.push_back(hardware_interface::CommandInterface(
+    get_node()->get_name(), std::string("linear/") + hardware_interface::HW_IF_VELOCITY,
+    &reference_interfaces_[0]));
+
+  reference_interfaces.push_back(hardware_interface::CommandInterface(
+    get_node()->get_name(), std::string("angular/") + hardware_interface::HW_IF_POSITION,
+    &reference_interfaces_[1]));
+
+  return reference_interfaces;
+}
+
+bool SteeringControllers::on_set_chained_mode(bool chained_mode)
+{
+  // Always accept switch to/from chained mode
+  return true || chained_mode;
+}
+
+controller_interface::CallbackReturn SteeringControllers::on_activate(
+  const rclcpp_lifecycle::State & /*previous_state*/)
+{
+  // Set default value in command
+  reset_controller_reference_msg(*(input_ref_.readFromRT()), get_node());
+
+  return controller_interface::CallbackReturn::SUCCESS;
+}
+
+controller_interface::CallbackReturn SteeringControllers::on_deactivate(
+  const rclcpp_lifecycle::State & /*previous_state*/)
+{
+  // TODO(anyone): depending on number of interfaces, use definitions, e.g., `CMD_MY_ITFS`,
+  // instead of a loop
+  for (size_t i = 0; i < nr_cmd_itfs_; ++i)
+  {
+    command_interfaces_[i].set_value(std::numeric_limits<double>::quiet_NaN());
+  }
+  return controller_interface::CallbackReturn::SUCCESS;
+}
+
+controller_interface::return_type SteeringControllers::update_reference_from_subscribers()
+{
+  auto current_ref = *(input_ref_.readFromRT());
```

Comment:
When we add support for ackermann_msgs we should also adjust this here.

Should we add ackerman_msgs support in a follow-up PR? We can do this on top of this already, but merge it separately.